nomad/structs/services.go

```diff
@@ -263,49 +282,58 @@ func (sc *ServiceCheck) TriggersRestarts() bool {
 // called.
 func (sc *ServiceCheck) Hash(serviceID string) string {
 	h := sha1.New()
-	io.WriteString(h, serviceID)
-	io.WriteString(h, sc.Name)
-	io.WriteString(h, sc.Type)
-	io.WriteString(h, sc.Command)
-	io.WriteString(h, strings.Join(sc.Args, ""))
-	io.WriteString(h, sc.Path)
-	io.WriteString(h, sc.Protocol)
-	io.WriteString(h, sc.PortLabel)
-	io.WriteString(h, sc.Interval.String())
-	io.WriteString(h, sc.Timeout.String())
-	io.WriteString(h, sc.Method)
-	// Only include TLSSkipVerify if set to maintain ID stability with Nomad <0.6
-	if sc.TLSSkipVerify {
-		io.WriteString(h, "true")
-	}
-
-	// Since map iteration order isn't stable we need to write k/v pairs to
-	// a slice and sort it before hashing.
-	if len(sc.Header) > 0 {
-		headers := make([]string, 0, len(sc.Header))
-		for k, v := range sc.Header {
-			headers = append(headers, k+strings.Join(v, ""))
-		}
-		sort.Strings(headers)
-		io.WriteString(h, strings.Join(headers, ""))
-	}
+	hashString(h, serviceID)
+	hashString(h, sc.Name)
+	hashString(h, sc.Type)
+	hashString(h, sc.Command)
+	hashString(h, strings.Join(sc.Args, ""))
+	hashString(h, sc.Path)
+	hashString(h, sc.Protocol)
+	hashString(h, sc.PortLabel)
+	hashBool(h, sc.Expose, "Expose")
```

Comment:
Since the Check.Expose field only exists in Nomad do we need to include it in the hash which is used for change detection? Aren't the generated Connect stanza changes sufficient to trigger the update?

I don't think it matters functionally either way, just curious if I'm understanding things.

Reply:
Ah that's a great point; yeah the `Hash` is only used for diffing against Consul. Within Nomad we use `Equals` and `diff.go`. I'll comment this out and leave a reminder.